pom.xml

```diff
@@ -24,15 +24,15 @@
   <parent>
     <groupId>org.apache</groupId>
     <artifactId>apache</artifactId>
-    <version>27</version>
+    <version>30</version>
```

Comment:
This is fine on its own probably, but I'd bet there are some plugin versions that can be updated at the same time, or whose versions are no longer needed to be specified as of the latest parent POM.

Reply:
I updated the puppycrawl version but that is the only additional change I could find given your suggestions.

Reply:
I didn't see any versions that could be removed, but I did bump some plugin versions and made some other follow-on improvements in 9cdafdb8c07d4e815b072aca70cf2ec0acc2df0c